esp-hal/ld/sections/rwtext.x

```diff
@@ -1,9 +1,13 @@
-
-
 SECTIONS {
   .rwtext : ALIGN(4)
   {
     . = ALIGN (4);
+    #IF ESP_HAL_PLACE_MORE_RODATA_IN_RAM
+      *(.rodata.*_esp_hal_internal_handler*)
+      *(.rodata.str.*)
+      *(.rodata.handle_interrupts)
+      *(.rodata..Lswitch.table.*)
```

Comment:
Is this double `..` correct?

Do we have any more granularity with what we include - is it possible to only include switch tables from the interrupt module perhaps?

Reply:
Shouldn't `handle_interrupts` already be in RAM? If not, why not, since it has the `#[ram]` attribute?

What is `_esp_hal_internal_handler`, I don't seem to be able to find that string in the source anywhere.

Reply:
Those look like this `.rodata..Lswitch.table._ZN16embassy_executor3raw20TaskStorage$LT$F$GT$4poll17h8bd1af7ee097c9d5E`, yes

We could try to wild-card match a bit more specific - the naming doesn't make that very easy (and I wonder how much this improves performance beyond ISRs - especially for any ... would be interesting)

Reply:
> the naming doesn't make that very easy

I _think_ modern linkers understand the rust demangling scheme; running that output through `rustfilt` gives me `.rodata..Lswitch.table.embassy_executor::raw::TaskStorage<F>::poll`. I suppose for esp-hal's interrupt module that might look like `.rodata..Lswitch.table.esp_hal::interrupt`, can we wildcard that?

> and I wonder how much this improves performance beyond ISRs - especially for any ... would be interesting

I do wonder too, but I'm thinking if we can minimize the ram used here, we might be able to enable this config by default - worst case 5~us interrupt latency is really nice.

Reply:
sure we can limit it here and maybe add another option to place all the switch tables in RAM - might be something which can improve performance in general (I still have no proof for that)

Reply:
> Shouldn't handle_interrupts already be in RAM? If not, why not, since it has the #[ram] attribute?

Yep - should - needs double checking why I added it (maybe by mistake or during desperately trying things)

> What is _esp_hal_internal_handler, I don't seem to be able to find that string in the source anywhere.

Created by the `#[handler]` macro and it's in DROM

`3c010020 3c010020       5c     4         C:\projects\esp\esp-hal\examples\target\riscv32imc-unknown-none-elf\release\deps\gpio_interrupt-d547762d74cfa5b6.gpio_interrupt.a9128762895f52da-cgu.0.rcgu.o:(.rodata._ZN14gpio_interrupt26__esp_hal_internal_handler17h8cf7e5aa7b7ed9b2E)
`

Reply:
had another look at the linker map file and (looking at wifi_embassy_dhcp) the amount of switch tables (and the size of those items) is not too much - what seems to be a lot is `.Lanon.*` and those are named like `.Lanon.fad58de7366495db4650cfefac2fcd61.137` - no indication where those are used unfortunately

Reply:
ok checked it

```text
check sizes "wifi_embassy_dhcp"

- baseline	0000068c	00001138
- all		000009a8	00005e10
- no Lswitch	0000068c	00005e10	(- 796)
- no Lanon	000009a8	00001358	(- 19128)
```

IMHO further qualifying the swtiches won't give us much other than headaches whenever we rename things - unfortunately we won't be able to further qualify Lanon